LiskHQ/lisk-desktop
# Setup side chain on local

### What was the problem?

This PR resolves #5173

### How was it solved?

- [x] Setup enevti core on jenkins
- [x] Setup enevti service on jenkins
- [x] Refactor lisk chain service, enevti chain service  and e2e test job procedure.
- [x] Ensure existing e2e test works

### How was it tested?

- Spin up core for both chains (i.e lisk and enevti) by running the following;
  - Open up a bash terminal and ensure you are in the `lisk-desktop` root directory, then run `CORE=lisk source e2e/scripts/run-core.sh` on that terminal. This should spin up lisk core on that terminal.
  - Open up a new bash terminal, while also still in the `lisk-desktop` root directory and run `CORE=enevti source e2e/scripts/run-core.sh`. This should spin up enevti core on that terminal.
  
- Spin up service for both enevti and lisk
  - Ensure you have docker available on your local machine.
  - Ensure you have cloned the lisk-service repo and you have separate copies of the service folder.
  - Name one of the above folders lisk-service and the other enevti-service
  - Open up a fresh new bash terminal and ensure you are in the `lisk-desktop` root directory then run `LISK_SERVICE_FILE_PATH="<relative path to the lisk-service folder>" GITHUB_APP_REGISTRY_REPO_BRANCH=jenkins-deployment sh e2e/scripts/run-service.sh` on that terminal. This should spin up a lisk-service that connects to the lisk-core initially created on that terminal.
  - Again, open up a new bash terminal and also ensure you are in the `lisk-desktop` root directory then run `ENEVTI_SERVICE_FILE_PATH="<relative path to the enveti service folder>" GITHUB_APP_REGISTRY_REPO_BRANCH=jenkins-deployment  source e2e/scripts/run-service.sh` on that terminal. This should spin up an `enevti-service` instance that connects to the enevti-core instance created initially on that terminal.
 
- With the above, you should have core and service for enevti and lisk up and running. Then you should be able to run the e2e test or add a local network connection to a running lisk-deskop application. With this, all endpoints should point to 127.0.0.1 and enevti should be part of the explorable applications if viewed from the lisk local main chain
